### PR TITLE
Add missing spanish translations

### PR DESCRIPTION
**Pull request by pedrodotmc**
#### Description of changes

- Add spanish translations to Authenticator dict.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/i18n/dictionaries/authenticator/es.ts [hunk 1]
```diff
@@ -14,10 +15,12 @@ export const esDict = {
   'Confirmation Code': 'Código de confirmación',
   Confirming: 'Confirmando',
   'Create a new account': 'Crear una cuenta nueva',
+  'Create Account': 'Crear cuenta',
   'Creating Account': 'Creando cuenta',
   Email: 'Email',
+  'Enter your email': 'Email',
   'Forgot Password': 'Olvidé mi contraseña',
-  'Incorrect username or password': 'Nombre de usuario o contraseña incorrecta',
+  'Incorrect username or password.': 'Nombre de usuario o contraseña incorrecta',
```
Comment (reesscot) [inline packages/ui/src/i18n/dictionaries/authenticator/es.ts:23]: Why add the period here? Won't this break the translation?

File: packages/ui/src/i18n/dictionaries/authenticator/es.ts [hunk 2]
```diff
@@ -14,10 +15,12 @@ export const esDict = {
   'Confirmation Code': 'Código de confirmación',
   Confirming: 'Confirmando',
   'Create a new account': 'Crear una cuenta nueva',
+  'Create Account': 'Crear cuenta',
   'Creating Account': 'Creando cuenta',
   Email: 'Email',
+  'Enter your email': 'Email',
```
Comment (reesscot) [inline packages/ui/src/i18n/dictionaries/authenticator/es.ts:21]: Shouldn't this be something like "Escriba su Email"?

**Review by reesscot — COMMENTED** [review]
Left some comments below

File: packages/ui/src/i18n/dictionaries/authenticator/es.ts (hunk 2 above)
Comment (pedrodotmc) [reply packages/ui/src/i18n/dictionaries/authenticator/es.ts:21]: Yes, let me fix it

File: packages/ui/src/i18n/dictionaries/authenticator/es.ts (hunk 1 above)
Comment (pedrodotmc) [reply packages/ui/src/i18n/dictionaries/authenticator/es.ts:23]: Currently, english version has that period
image
this translation is not working

**Review by wlee221 — DISMISSED** [review]
Looks good based on my 4 years of high school Spanish that I all forgot, but I'll err on @reesscot to give the final approval ;)

**Review by wlee221 — APPROVED** [review]

**Pull request merged**